examples/encrypted-communications/1.js

```diff
@@ -1,63 +1,56 @@
 'use strict'
 
-const libp2p = require('../../')
+const Libp2p = require('../../')
 const TCP = require('libp2p-tcp')
-const SPDY = require('libp2p-spdy')
+const Mplex = require('libp2p-mplex')
 const SECIO = require('libp2p-secio')
 const PeerInfo = require('peer-info')
-const waterfall = require('async/waterfall')
-const parallel = require('async/parallel')
-const pull = require('pull-stream')
-const defaultsDeep = require('@nodeutils/defaults-deep')
-
-class MyBundle extends libp2p {
-  constructor (_options) {
-    const defaults = {
-      modules: {
-        transport: [ TCP ],
-        streamMuxer: [ SPDY ],
-        connEncryption: [ SECIO ]
-      }
-    }
 
-    super(defaultsDeep(_options, defaults))
-  }
-}
+const pipe = require('it-pipe')
+const { map } = require('streaming-iterables')
+const { toBuffer } = require('it-buffer')
 
-function createNode (callback) {
-  let node
-
-  waterfall([
-    (cb) => PeerInfo.create(cb),
-    (peerInfo, cb) => {
-      peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
-      node = new MyBundle({
-        peerInfo
-      })
-      node.start(cb)
+const createNode = async () => {
+  const peerInfo = await PeerInfo.create()
+  peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
+
+  const node = await Libp2p.create({
+    peerInfo,
+    modules: {
+      transport: [TCP],
+      streamMuxer: [Mplex],
+      connEncryption: [SECIO]
     }
-  ], (err) => callback(err, node))
-}
+  })
 
-parallel([
-  (cb) => createNode(cb),
-  (cb) => createNode(cb)
-], (err, nodes) => {
-  if (err) { throw err }
+  await node.start()
 
-  const node1 = nodes[0]
-  const node2 = nodes[1]
+  return node
+}
 
-  node2.handle('/a-protocol', (protocol, conn) => {
-    pull(
-      conn,
-      pull.map((v) => v.toString()),
-      pull.log()
+;(async () => {
+  const [node1, node2] = await Promise.all([
+    createNode(),
+    createNode()
+  ])
+
+  node2.handle('/a-protocol', ({ stream }) => {
+    pipe(
+      stream,
+      toBuffer,
+      map(String),
+      source => (async function () {
+        for await (const msg of source) {
+          console.log(msg)
+        }
+      })(),
     )
```

Comment:
You can `toString` a BufferList, so this could be simplified. Also, the sink function is more complex than it needs to be:
```js
pipe(
  stream,
  async function (source) {
    for await (const msg of source) {
      console.log(msg.toString())
    }
  }
)
```

Reply:
I tried to mimic what was before, but considering that decision again, I think we should simplify :)

Reply:
changed